sdk/cosmosdb/cosmos/src/auth.ts

```diff
@@ -62,6 +62,17 @@ export async function setAuthorizationHeader(
     headers[Constants.HttpHeaders.Authorization] = encodeURIComponent(
       await clientOptions.tokenProvider({ verb, path, resourceId, resourceType, headers })
     );
+  } else if (clientOptions.aadCredentials) {
+    //breaking: throw if aadCredentials property is missing function is removed or type changes
+    if (typeof clientOptions.aadCredentials?.getToken !== "function") {
+      throw new Error(
+        "Error: function `getToken` is missing on AAD credentials. dependancy: @azure/identity"
+      );
+    }
+    const token = await clientOptions.aadCredentials.getToken(clientOptions.endpoint);
```

Comment:
https://github.com/Azure/azure-sdk-for-js/blob/82a9b0a30356813101a1ec076489cdde7f04ef93/sdk/core/core-auth/src/tokenCredential.ts#L14

Token generation is expensive and documented that caching and refreshing needs to handle by the caller explicitly.

Reply:
this is how JS handles token generation.